packages/repository-json-schema/src/build-schema.ts

```diff
@@ -467,7 +467,9 @@ export function modelToJsonSchema<T extends object>(
       delete schema.definitions;
     }
 
-    result.definitions[name] = schema;
+    if (result !== schema) {
```

Comment:
IIUC, because this check is executed after we initialize `result.definitions`, models with circular references may end up with an empty object in `definitions` property, which I find a bit ugly.

Let's try to find a way how to avoid having `definitions` set to an empty object.